README.md

```diff
@@ -140,6 +140,51 @@ cursor_before = metadata.before
 IO.puts "total count: #{metadata.total_count}"
 ```
 
+## Dynamic expressions
+
+```elixir
+  query =
+    from(
+      f in Post,
+      # Alias for fragment must match witch cursor field name in fetch_cursor_value_fun and cursor_fields
+      select_merge: %{
+        rank_value:
+          fragment("ts_rank(document, plainto_tsquery('simple', ?)) AS rank_value", ^q)
+      },
+      where: fragment("document @@ plainto_tsquery('simple', ?)", ^q),
+      order_by: [
+        desc: fragment("rank_value"),
+        desc: f.id
+      ]
+    )
+    query
+    |> Repo.paginate(
+      limit: 30,
+      fetch_cursor_value_fun: fn
+        # Here we build the rank_value for each returned row
+        schema, :rank_value ->
+          {:ok, %{rows: [[rank_value]]}} =
+            Repo.query("SELECT ts_rank($1, plainto_tsquery('simple', $2))", [
+              schema.document,
+              q
+            ])
+          rank_value
+        schema, field ->
+          Paginator.default_fetch_cursor_value(schema, field)
+      end,
+      cursor_fields: [
```

Comment:
Could I suggest adding an example here for when an explicit ordering needs to be defined? Like

```
        {{:rank_value, # Here we build the rank_value that will be used in the where clause
         fn ->
           dynamic(
             [x],
             fragment("ts_rank(document, plainto_tsquery('simple', ?))", ^q)
           )
         end}, :desc},
```

Reading the docs as they are, this wasn't immediately clear to me, I had to dig through the code a bit.

Thanks!

Reply:
@marinakr yes, it does work for me. Can you paste your calling code here? For example, here's what my use case looks like:

```
        user_position_geom =
          user
          |> User.with_address()
          |> get_in([:address, :position_geom])

       # Query.events_for_user/2 just returns all events that a user is eligible to participate in.
       # It also adds :events and :location_address aliases to the query, which you'll see referenced in the cursor_fields.
        user
        |> Query.events_for_user(params)
        |> Repo.paginate(
          after: params[:search_after],
          include_total_count: true,
          limit: 20,
          cursor_fields: [
            {{:start_date,
              fn ->
                dynamic(
                  [events: e, location: l],
                  fragment("(? at time zone 'UTC' at time zone ?)::date", e.starts_at, l.timezone)
                )
              end}, :asc},
            {{:distance,
              fn ->
                dynamic(
                  [location_address: la],
                  Geo.PostGIS.st_distance_in_meters(la.position_geom, ^user_position_geom)
                )
              end}, :asc},
            {{:start_time,
              fn ->
                dynamic(
                  [events: e, location: l],
                  fragment("(? at time zone 'UTC' at time zone ?)::time", e.starts_at, l.timezone)
                )
              end}, :asc},
            ends_at: :asc,
            id: :asc
          ]
        )
```

The goal of this query being to paginate the results based on their starting date, then their distance from the user, and then their start time.

Reply:
@darrenklein the issue appears if paginate it with cursor_after
`deps/paginator/lib/paginator/cursor.ex:9`
code
```
|> Plug.Crypto.non_executable_binary_to_term([:safe])
```
Raise an error if paginate with after cursor

Code exapmle:
```
[
      {:vendors,
       fn ->
         dynamic([vendors: v], fragment("LOWER(?)", v.name))
       end},
      :id
    ]
  ```
  Works good on first iteration - however, if paginate it later with `after` cursor
  Actually I should have had noticed this before:
  https://github.com/duffelhq/paginator/pull/166/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R190
  
  To avoid it, I think something like {:mfa, Mod, :fn, []} should be passed and handled in paginator
  Because of `Paginator.Cursor` that decodes only `non_executable_binary_to_term` I don't see how to add fragment in  `where` condition

Reply:
@marinakr Hm, I'm sorry, I don't know any more about this, really. I'd say maybe post about this in the "Issues" section of the main repo?

Slava Ukraini 👊